plugins/optimization-detective/storage/rest-api.php

```diff
@@ -173,24 +176,24 @@ function od_handle_rest_request( WP_REST_Request $request ) {
 	 *
 	 * @since 0.6.0
 	 *
-	 * @param array $context {
+	 * @param array                                   $context {
 	 *     Context about the successful URL Metric collection.
 	 *
-	 *     @var int                                   $post_id
-	 *     @var WP_REST_Request<array<string, mixed>> $request
-	 *     @var OD_Strict_URL_Metric                  $url_metric
-	 *     @var OD_URL_Metrics_Group                  $group
-	 *     @var OD_URL_Metrics_Group_Collection       $group_collection
+	 *     @type int                                   $post_id                     ID for URL metrics post.
+	 *     @type WP_REST_Request<array<string, mixed>> $request                     Storage request.
+	 *     @type OD_Strict_URL_Metric                  $url_metric                  URL metric.
+	 *     @type OD_URL_Metric_Group                   $url_metric_group            URL metric group.
+	 *     @type OD_URL_Metric_Group_Collection        $url_metric_group_collection URL metric group collection.
 	 * }
 	 */
 	do_action(
 		'od_url_metric_collected',
 		array(
-			'post_id'                      => $post_id,
-			'request'                      => $request,
-			'url_metric'                   => $url_metric,
-			'url_metrics_group'            => $group,
-			'url_metrics_group_collection' => $group_collection,
+			'post_id'                     => $post_id,
+			'request'                     => $request,
+			'url_metric'                  => $url_metric,
+			'url_metric_group'            => $url_metric_group,
+			'url_metric_group_collection' => $url_metric_group_collection,
 		)
```

Comment:
Instead of using an array here, it would be better to implement something like `OD_Tag_Visitor_Context` which has properties instead of array keys. This would allow for autocompletion as well as it would support possible future renaming of the members.

Reply:
Like this: 1cf24f17f5f06706d3cc4db7586de1a4adb3b071